lib/arena.rb

```diff
@@ -1,2 +1,68 @@
+require "pry"
+require_relative "../lib/gladiator"
+
 class Arena
+  attr_accessor :name, :gladiators, :battle, :gladiator_names
+
+  def initialize(name, gladiators = [])
+    @name = name.capitalize
+    @gladiators = gladiators
+    @battle = []
+    @gladiator_names = []
+  end
+
+  def add_gladiator(gladiator)
+    gladiators << gladiator unless gladiators.length == 2
+  end
+
+  def fight
+    unless gladiators.length < 2
+      gladiators.each do |gladiator|
+          battle << gladiator.weapon
+          gladiator_names << gladiator.name
+      end
+      get_winner(battle)
+    end
+  end
+
+  def get_winner(battle)
+    if gladiator_names.include?("Maximus")
+      gladiators.delete_if {|gladiator| gladiator.name != "Maximus"} if thumbs_up == "d"
+    elsif battle.sort! == ["Spear", "Trident"]
+      gladiators.delete_if {|gladiator| gladiator.weapon == "Spear"} if thumbs_up == "d"
+    elsif battle.sort! == ["Club", "Spear"]
+      gladiators.delete_if {|gladiator| gladiator.weapon == "Club"} if thumbs_up == "d"
+    elsif battle.sort! == ["Club", "Trident"]
```

Comment:
some solid ruby here Pat! How would you go about abstracting some of the repetitive code and checks?